jstz_cli/src/debug_api/account.rs

```diff
@@ -0,0 +1,141 @@
+use std::ops::Deref;
+
+use boa_engine::{
+    js_string, object::ObjectInitializer, Context, JsArgs, JsNativeError, JsObject,
+    JsResult, JsValue, NativeFunction,
+};
+use jstz_core::{host_defined, kv::Transaction, runtime};
+use jstz_crypto::public_key_hash::PublicKeyHash;
+use jstz_proto::context::account::Account;
+
+macro_rules! preamble {
+    ($args:ident, $context:ident, $tx:ident) => {
+        host_defined!($context, host_defined);
+        let mut $tx = host_defined
+            .get_mut::<Transaction>()
+            .expect("Curent transaction undefined");
+    };
+}
+
+fn get_public_key_hash(account: &str) -> JsResult<PublicKeyHash> {
+    PublicKeyHash::from_base58(account).map_err(|_| {
+        JsNativeError::typ()
+            .with_message("Could not parse the address.")
+            .into()
+    })
+}
+
+pub struct AccountApi;
+
+impl AccountApi {
+    fn balance(
+        _this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        preamble!(args, context, tx);
+
+        let account: String = args.get_or_undefined(0).try_js_into(context)?;
+
+        let pkh = get_public_key_hash(account.as_str())?;
+
+        let result =
+            runtime::with_global_host(|rt| Account::balance(rt.deref(), &mut tx, &pkh))?;
+
+        if result <= i32::MAX as u64 {
+            Ok(JsValue::from(result as i32))
+        } else {
+            Err(JsNativeError::typ().with_message("Balance overflow").into())
+        }
+    }
+
+    fn set_balance(
+        _this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        preamble!(args, context, tx);
+
+        let account: String = args.get_or_undefined(0).try_js_into(context)?;
+
+        let balance: u64 = args.get_or_undefined(1).try_js_into(context)?;
+
+        let pkh = get_public_key_hash(account.as_str())?;
+
+        if balance <= i32::MAX as u64 {
+            runtime::with_global_host(|rt| {
+                Account::set_balance(rt.deref(), &mut tx, &pkh, balance)
+            })?;
+            Ok(JsValue::undefined())
+        } else {
+            Err(JsNativeError::typ().with_message("Balance overflow").into())
+        }
```

Comment:
Why are you checking against `i32::MAX`, it is entirely possible for boa to represent an integer using `JsValue::Rational` which can have a larger value.